scheduler/src/cook/mesos/util.clj

```diff
@@ -24,7 +24,59 @@
             [metatransaction.core :refer (db)]
             [metrics.timers :as timers]
             [plumbing.core :as pc :refer (map-vals map-keys)])
-  (:import [java.util Date]))
+    (:import
+      [com.google.common.cache Cache]
+      [com.google.common.cache CacheBuilder]
+      [java.util.concurrent TimeUnit]
+      [java.util Date]))
+
+
+(defn new-cache []
+  "Build a new cache"
+  (-> (CacheBuilder/newBuilder)
+      (.maximumSize 1000000)
+      ;; if its not been accessed in 2 hours, whatever is going on, its not being visted by the
+      ;; scheduler loop anymore. E.g., its probably failed/done and won't be needed. So,
+      ;; lets kick it out to keep cache small.
+      (.expireAfterAccess 2 TimeUnit/HOURS)
+      (.build)))
+
+(defn lookup-cache!
+      "Generic cache. Caches under a key (extracted from the item with extract-key-fn. Uses miss-fn to fill
+      any misses. Caches only postive hits where both functions return non-nil"
+      [cache-atom extract-key-fn miss-fn item]
```

Comment:
Nitpick: `extract-key-fn` could not be an argument in this function, the caller can send the extracted key.

Reply:
It works better this way, as re-using extract-key-fn can be reused across multiple caches. What you're asking for is essentially to inline the invocation of this into all of the callers, versus passing it.